src/sentry/profiles/task.py

```diff
@@ -365,6 +424,25 @@ def truncate_stack_needed(
         ) -> List[Any]:
             return stack
 
+    symbolicated_frames = stacktraces[0]["frames"]
+
+    if len(frames_sent) > 0:
+        raw_frames = profile["profile"]["frames"]
+        new_frames = []
+        raw_frame_idx = 0
+        frames_dict = get_frame_index_map(symbolicated_frames)
```

Comment:
Nit: You might be able to avoid calling `get_frame_index_map` twice (once here and once below).